### PR TITLE
[TwigBundle][FrameworkBundle] Remove the internals from debug autowiring

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no
| Deprecations? |no 
| Tests pass?   | yes
| Fixed tickets | #24986
| License       | MIT
| Doc PR        |

#SymfonyConHackday2017
@nicolas-grekas @weaverryan @fabpot @stof It should be OK to review and to merge.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Test don't pass yet :)
Wondering two things:

- does ConsoleCommandPass prefer a specific naming scheme?
- running debug:autowiring on a 3.4 standard edition, I get this list. Looks like there could me more FQCN ids removed (not only on symfony/symfony, would you mind cleaning them also?)

```
  Sensio\Bundle\FrameworkExtraBundle\EventListener\ControllerListener                 
  Sensio\Bundle\FrameworkExtraBundle\EventListener\HttpCacheListener                  
  Sensio\Bundle\FrameworkExtraBundle\EventListener\IsGrantedListener                  
  Sensio\Bundle\FrameworkExtraBundle\EventListener\ParamConverterListener             
  Sensio\Bundle\FrameworkExtraBundle\EventListener\SecurityListener                   
  Sensio\Bundle\FrameworkExtraBundle\EventListener\TemplateListener                   
  Sensio\Bundle\FrameworkExtraBundle\Request\ArgumentNameConverter                    
  Sensio\Bundle\FrameworkExtraBundle\Request\ParamConverter\DateTimeParamConverter    
  Sensio\Bundle\FrameworkExtraBundle\Request\ParamConverter\DoctrineParamConverter    
  Sensio\Bundle\FrameworkExtraBundle\Request\ParamConverter\ParamConverterManager     
  Sensio\Bundle\FrameworkExtraBundle\Routing\AnnotatedRouteControllerLoader           
  Sensio\Bundle\FrameworkExtraBundle\Security\ExpressionLanguage                      
  Sensio\Bundle\FrameworkExtraBundle\Templating\TemplateGuesser                       
  Symfony\Bridge\Twig\Command\DebugCommand                                            
  Symfony\Bundle\FrameworkBundle\Command\AboutCommand                                 
  Symfony\Bundle\FrameworkBundle\Command\AssetsInstallCommand                         
  Symfony\Bundle\FrameworkBundle\Command\CacheClearCommand                            
  Symfony\Bundle\FrameworkBundle\Command\CachePoolClearCommand                        
  Symfony\Bundle\FrameworkBundle\Command\CachePoolPruneCommand                        
  Symfony\Bundle\FrameworkBundle\Command\CacheWarmupCommand                           
  Symfony\Bundle\FrameworkBundle\Command\ConfigDebugCommand                           
  Symfony\Bundle\FrameworkBundle\Command\ConfigDumpReferenceCommand                   
  Symfony\Bundle\FrameworkBundle\Command\ContainerDebugCommand                        
  Symfony\Bundle\FrameworkBundle\Command\DebugAutowiringCommand                       
  Symfony\Bundle\FrameworkBundle\Command\EventDispatcherDebugCommand                  
  Symfony\Bundle\FrameworkBundle\Command\RouterDebugCommand                           
  Symfony\Bundle\FrameworkBundle\Command\RouterMatchCommand                           
  Symfony\Bundle\FrameworkBundle\Command\XliffLintCommand                             
  Symfony\Bundle\FrameworkBundle\Command\YamlLintCommand                              
  Symfony\Bundle\FrameworkBundle\Controller\RedirectController                        
  Symfony\Bundle\FrameworkBundle\Controller\TemplateController                        
  Symfony\Bundle\SecurityBundle\Command\UserPasswordEncoderCommand                    
  Symfony\Bundle\TwigBundle\Command\LintCommand                                       
  Symfony\Component\Config\Resource\SelfCheckingResourceChecker                       
  Symfony\Component\DependencyInjection\Config\ContainerParametersResourceChecker     
  Symfony\Component\ExpressionLanguage\ExpressionLanguage                             
  Symfony\Component\Form\Command\DebugCommand                                         
  Symfony\Component\Routing\Loader\AnnotationDirectoryLoader                          
  Symfony\Component\Routing\Loader\AnnotationFileLoader                               
```

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/console.xml [hunk 1]
```diff
@@ -13,95 +13,95 @@
...
             <tag name="console.command" command="lint:xliff" />
         </service>
 
-        <service id="Symfony\Bundle\FrameworkBundle\Command\YamlLintCommand">
+        <service id="console.yaml_lint" class="Symfony\Bundle\FrameworkBundle\Command\YamlLintCommand">
             <tag name="console.command" command="lint:yaml" />
         </service>
 
-        <service id="Symfony\Component\Form\Command\DebugCommand">
+        <service id="console.debug_command" class="Symfony\Component\Form\Command\DebugCommand">
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/console.xml]: suffix command here, but not on the other ones, should it be always here?

File: src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTest.php [hunk 2]
```diff
@@ -314,7 +313,7 @@ public function testWorkflowServicesCanBeEnabled()
         $container = $this->createContainerFromFile('workflows_enabled');
 
         $this->assertTrue($container->has(Registry::class));
-        $this->assertTrue($container->hasDefinition(WorkflowDumpCommand::class));
+        $this->assertTrue($container->hasDefinition('console.workflow_dump'));
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTest.php]: `console.command.workflow_dump`? other convention

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/routing.xml [hunk 3]
```diff
@@ -112,13 +112,13 @@
             <argument>%kernel.debug%</argument>
         </service>
 
-        <service id="Symfony\Bundle\FrameworkBundle\Controller\RedirectController" public="true">
+        <service id="framework.controller.redirect" class="Symfony\Bundle\FrameworkBundle\Controller\RedirectController" public="true">
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/routing.xml]: I think we need to keep this and the next one, not for autowiring, but to ease routing

**Comment by Simperfit** [comment]
The review has been taken into account.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/console.xml [hunk 4]
```diff
@@ -7,101 +7,101 @@
...
             <tag name="console.command" command="lint:xliff" />
         </service>
 
-        <service id="Symfony\Bundle\FrameworkBundle\Command\YamlLintCommand">
+        <service id="console.command.yaml_lint" class="Symfony\Bundle\FrameworkBundle\Command\YamlLintCommand">
             <tag name="console.command" command="lint:yaml" />
         </service>
 
-        <service id="Symfony\Component\Form\Command\DebugCommand">
+        <service id="console.command.debug" class="Symfony\Component\Form\Command\DebugCommand">
```
Comment (ro0NL) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/console.xml]: `console.command.form_debug` id say

File: src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTest.php (hunk 2 above)
Comment (ro0NL) [reply src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTest.php]: `console.command.<underscored_name>` :+1:

File: src/Symfony/Bundle/TwigBundle/Resources/config/console.xml [hunk 5]
```diff
@@ -7,13 +7,13 @@
     <services>
         <defaults public="false" />
 
-        <service id="Symfony\Bridge\Twig\Command\DebugCommand">
+        <service id="twig.debug_command" class="Symfony\Bridge\Twig\Command\DebugCommand">
```
Comment (ro0NL) [inline src/Symfony/Bundle/TwigBundle/Resources/config/console.xml]: `twig.commmand.*`?

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/console.xml [hunk 6]
```diff
@@ -7,101 +7,101 @@
     <services>
         <defaults public="false" />
 
-        <service id="console.error_listener" class="Symfony\Component\Console\EventListener\ErrorListener">
+        <service id="console.command.error_listener" class="Symfony\Component\Console\EventListener\ErrorListener">
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/console.xml]: renamig is not good. This service is not a command

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/console.xml (hunk 4 above)
Comment (stof) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/console.xml]: agreed

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/console.xml [hunk 7]
```diff
@@ -7,101 +7,101 @@
...
+        <service id="console.command.translation_update" class="Symfony\Bundle\FrameworkBundle\Command\TranslationUpdateCommand">
             <argument type="service" id="translation.writer" />
             <argument type="service" id="translation.reader" />
             <argument type="service" id="translation.extractor" />
             <argument>%kernel.default_locale%</argument>
             <tag name="console.command" command="translation:update" />
         </service>
 
-        <service id="Symfony\Bundle\FrameworkBundle\Command\WorkflowDumpCommand">
+        <service id='console.workflow_dump' class="Symfony\Bundle\FrameworkBundle\Command\WorkflowDumpCommand">
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/console.xml]: missing `.command` for consistency

File: src/Symfony/Bundle/TwigBundle/Resources/config/console.xml (hunk 5 above)
Comment (stof) [reply src/Symfony/Bundle/TwigBundle/Resources/config/console.xml]: would make sense for consistency.

**Review by stof — CHANGES_REQUESTED** [review]

**Comment by chalasr** [comment]
> does ConsoleCommandPass prefer a specific naming scheme?

@nicolas-grekas if that's what you wonder, AddConsoleCommandPass should process correctly no matter the service id

**Comment by Simperfit** [comment]
done @stof

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/console.xml [hunk 8]
```diff
@@ -13,95 +13,95 @@
...
+        <service id="console.command.translation_update" class="Symfony\Bundle\FrameworkBundle\Command\TranslationUpdateCommand">
             <argument type="service" id="translation.writer" />
             <argument type="service" id="translation.reader" />
             <argument type="service" id="translation.extractor" />
             <argument>%kernel.default_locale%</argument>
             <tag name="console.command" command="translation:update" />
         </service>
 
-        <service id="Symfony\Bundle\FrameworkBundle\Command\WorkflowDumpCommand">
+        <service id='console.command.workflow_dump' class="Symfony\Bundle\FrameworkBundle\Command\WorkflowDumpCommand">
```
Comment (dunglas) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/console.xml]: double quote

**Review by dunglas — CHANGES_REQUESTED** [review]

**Comment by nicolas-grekas** [comment]
There are two remaining service when testing this with the standard edition:
- Symfony\Component\Routing\Loader\AnnotationDirectoryLoader
- Symfony\Component\Routing\Loader\AnnotationFileLoader

looking more precisely with @Simperfit, we discovered that these classes are actually registered twice: first with a class as id, second with a regular id. Needs to be understood and fixed.

**Review by nicolas-grekas — APPROVED** [review]
(the remaining two services are declared by SensioFrameworkExtraBundle)

**Comment by nicolas-grekas** [comment]
Thank you @Simperfit.

**Pull request merged**